streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1243,9 +1243,7 @@ public String toString(final String indent) {
 
     public Map<MetricName, Metric> adminClientMetrics() {
         final Map<MetricName, ? extends Metric> adminClientMetrics = taskManager.adminClient().metrics();
-        final LinkedHashMap<MetricName, Metric> result = new LinkedHashMap<>();
-        result.putAll(adminClientMetrics);
-        return result;
+        return new LinkedHashMap<>(adminClientMetrics);
```

Comment:
Side cleanup

Reply:
Why do we need a LinkedHashMap here? Particularly because the interface doesn't suggest it needs anything more than a Map implementation? Actually I'm also wondering why we need to copy at all since we get back a read only view of the map from Admin and it appears the only consumer of adminClientMetrics immediately copies the results anyway.

Reply:
Not sure about the `LinkedHashMap` -- maybe @guozhangwang knows more?

For why to copy: it's not a public contract that we get a read-only map and hence we need to guard the map from writes via a copy. Not sure if we could rely on an AdminClient implementation detail... \cc @cmccabe ?

Reply:
Yes, the only use case I found above this code is to immediately copy it into yet another map (which is why the LinkedHashMap also doesn't make sense), so it should not matter, unless we're expecting to do something else with it down the road. Anyway, per our discussion I'm OK with this going in without fixing it, but it is low hanging fruit to reduce unnecessary work.